libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,240 @@
+  ) {
+    const { host, port, tls, caching, cacheTtl, maxRetries, maxDelay, eventStreaming } = getOptions(options);
+    const transport = createConnectTransport({
+      baseUrl: `${tls ? 'https' : 'http'}://${host}:${port}`,
+    });
+    this._promiseClient = promiseClient ? promiseClient : createPromiseClient(Service, transport);
+    this._callbackClient = callbackClient ? callbackClient : createCallbackClient(Service, transport);
+    this._cache = cache ? cache : new LocalStorageFlagdCache({ itemTtl: cacheTtl }, logger);
+    this._maxRetries = maxRetries === 0 ? Infinity: maxRetries;
+    this._maxDelay = maxDelay;
+    this._streamingEnabled = eventStreaming;
+    this._cacheEnabled = caching;
+    this._logger = logger;
+
+    if (this._streamingEnabled) {
+      this.retryConnect();
+    } else {
+      this._connected = true;
+    }
+  }
+
+  // if the cache is active, or we're connected, or streaming is disabled, we're ready.
+  get ready() {
+    return this._connected || this._cacheEnabled || !this._streamingEnabled;
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<boolean>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveBoolean);
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<string>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveString);
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<number>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveFloat);
+  }
+
+  resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<U>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveObject);
+  }
+
+  private async evaluate<T extends FlagValue>(
+    flagKey: string,
+    defaultValue: T,
+    context: EvaluationContext,
+    resolver: (
+      request: PartialMessage<ResolveAnyRequest>,
+      options?: CallOptions | undefined
+    ) => Promise<ResolveAnyResponse>
+  ): Promise<ResolutionDetails<T>> {
+    try {
+      const result = await this._cache.get<T>(flagKey, context);
+      if (this._cacheEnabled && result) {
+        return result;
+      }
+      const res = await resolver({flagKey, context: Struct.fromJson(context as JsonValue)});
+      const details = {
+        // if this is a Struct (object resolution), parse it
+        value: (res.value && res.value instanceof Struct ? res.value.toJson() : res.value) as unknown as T,
+        reason: res.reason,
+        variant: res.variant,
+      };
+      if (this._cacheEnabled) {
+        this._cache.set(flagKey, context, details);
```

Comment:
The provider will default. I went back and forth on whether or not the provider should default if the caching fails. I went this way because if the cache is failing and you configured it, you probably want to consider it a significant error, and so we should behave as such (default). I'm not sure if the caching is more of an "besides-the-point" concern though. Interested in your input.

Reply:
We aren't awaiting this (which is fine) but errors would be unhandled.

What do you think about something like this?

```typescript
this._cache.set(flagKey, context, details).catch((err) => this._logger?.error(err))
```

Reply:
Fixed as we discussed.